rest_framework/relations.py

```diff
@@ -169,6 +169,12 @@ def get_queryset(self):
             # for the field.
             # Eg: 'MyRelationship(queryset=ExampleModel.objects.all())'
             queryset = queryset.all()
+        if hasattr(self.parent, "Meta") and hasattr(self.parent.Meta, "model"):
+            queryset = queryset.filter(
+                **self.parent.Meta.model._meta.get_field(
+                    self.source
+                ).get_limit_choices_to()
+            )
```

Comment:
Maybe we could skip the filter if there's no `limit_choices_to` ?
Not sure how to do that though